src/kernel/src/namedobj/mod.rs

```diff
@@ -11,17 +11,20 @@ impl NamedObjManager {
     pub fn new(sys: &mut Syscalls) -> Arc<Self> {
         let namedobj = Arc::new(Self {});
 
-        sys.register(557, &namedobj, Self::sys_namedobj_create);
+        sys.register(557, &Arc::clone(&namedobj), Self::sys_namedobj_create);
```

Comment:
tf is this? :eyes:

Reply:
Without cloning, it sort of panics in the register-

Reply:
@SuchAFuriousDeath If I can find a way to make namedobj_create a public function, then I'll move mdbg.

Reply:
> Without cloning, it sort of panics in the register-

I don't believe that. Everywhere else it's fine.

Reply:
Nevermind, I think I did that as a pseudo fix to an earlier bug... Seems to work fine without cloning.